src/lib.rs

```diff
@@ -54,6 +57,13 @@ pub enum Machine {
     X86_64,
 }
 
+/// Information from an object file that can be used to locate separate debug info.
+#[derive(Debug, Clone, Copy)]
+pub enum DebugFileInfo {
+    /// The UUID from a Mach-O `LC_UUID` load command.
+    MachOUuid(Uuid),
```

Comment:
I wonder if it makes sense to have this be an associated type of the trait, something like:

```rust
impl Object for MachOFile {
    type DebugFileInfo = Uuid;

    fn get_debug_file_info(&self) -> Option<Uuid> { ... }
}
```

And then when we add support for `GnuDebugLink` for elf files it could have its own associated type as well.

For object files that don't have separate debug info, or we haven't implemented it yet, then we can use `()` as the associated type.

Thoughts on this approach?

Reply:
I'm not sure that returning an associated type is enough. The goal of the `Object` trait is to allow you to write cross-platform code that optionally allows you to only include support for the native platform. For `debug_file_info`, using an associated type for the return value is more useful if the return value can also be handled in cross-platform code, and the way we've handled that for other associated types is for each of them to satisfy a trait too. So the question is, what trait should this return value satisfy? And I don't have any good ideas for that.

Maybe the API should instead be `fn debug_file_path(&self) -> Option<PathBuf>`, and the implementations of this would use the `moria` crate (so `moria` would need to become independent of `object`).

Reply:
It gets a little twisty, certainly. I wasn't sure which way the dependency ought to go, just that I wanted a straightforward API for users. Having the APIs in `object` continue to be platform-agnostic from the outside seems sensible. One thing to consider is that `moria` is going to wind up dragging in a bunch of other dependencies because of what it wants to accomplish. In the future I might want to add APIs to locate debug symbols by fetching them from symbol servers over HTTP, which is going to pull in a whole world of things. Do we want all of that as dependencies on `object`, or would it be better to keep `object` simpler and have `moria` as an optional crate people can pull in to get that functionality?

The other way we can go with this is to provide a method on `File` that returns what's currently the `FileInternal` enum, to allow callers to delve into format-specific implementation details. Then we could just have methods like `Mach::get_uuid` and `Elf::get_build_id` which would be very straightforward.

Reply:
I don't think that exposing `FileInternal` would be any better for cross-platform code than the `DebugFileInfo` enum, and enum would be easier to use.

We do want to keep object dependencies simple. Even if we added `debug_file_path`, we would still need to expose the UUID for the more complex uses. So let's leave it how it is.